code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/Utils.java

```diff
@@ -176,4 +179,12 @@ static Map<String, String> extractQueryParameters(final String queryString) {
 
         return parameters;
     }
+
+    static boolean isInAppMessageEvent(final Event event) {
+        final Map<String, Object> consequenceMap = DataReader.optTypedMap(Object.class, event.getEventData(), CampaignConstants.EventDataKeys.RuleEngine.CONSEQUENCE_TRIGGERED, null);
+        if (MapUtils.isNullOrEmpty(consequenceMap)) {
+            return false;
+        }
+        return consequenceMap.get(CampaignConstants.EventDataKeys.RuleEngine.MESSAGE_CONSEQUENCE_TYPE).equals(CampaignConstants.MESSAGE_CONSEQUENCE_MESSAGE_TYPE);
```

Comment:
Can we switch the comparison here
onsequenceMap.get(CampaignConstants.EventDataKeys.RuleEngine.MESSAGE_CONSEQUENCE_TYPE) can be null .. so .equals on it might fail with NPE

It's better to have
CampaignConstants.MESSAGE_CONSEQUENCE_MESSAGE_TYPE.equals(consequenceMap.get)

Reply:
i've reversed the comparison to make it safe for null values